service/src/requestSchemas.ts

```diff
@@ -3,6 +3,7 @@ import { z } from 'zod';
 import { BadRequestError, NotImplementedError } from './util/errorUtils';
 
 const DATE_REGEX = /([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)(-(0[1-9]|1[0-2])(-(0[1-9]|[1-2][0-9]|3[0-1]))?)/;
+const VERSION_REGEX = /^\d+\.\d+\.\d+\.\d+$/;
```

Comment:
This regex makes it require the full semantic version including the 4th "revision" spot. It is implied that is required in $draft definition but elsewhere in CRMI they use only 3 parts to the "patch" spot. This should allow for the revision spot to be optional.

```suggestion
const VERSION_REGEX = /^\d+\.\d+\.\d+(\.\d+|)$/;
```